controllers/clusterresources/postgresqluser_controller.go

```diff
@@ -231,40 +292,108 @@ func (r *PostgreSQLUserReconciler) createUser(
+					return fmt.Errorf("cannot create PostgreSQL firewall rule, err: %w", err)
 				}
 			}
 		}
 	}
 
+	createUserQuery := fmt.Sprintf(`CREATE USER "%s" WITH PASSWORD '%s'`, newUserName, newPassword)
+	err = r.ExecPostgreSQLQuery(ctx, createUserQuery, defaultCreds, clusterName, defaultUserSecretNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *PostgreSQLUserReconciler) deleteUser(
+	ctx context.Context,
+	newUserName string,
+	defaultUserSecretNamespacedName clusterresourcesv1beta1.NamespacedName,
+) error {
+	defaultCreds, clusterName, err := r.getDefaultPostgreSQLUserCreds(ctx, defaultUserSecretNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	deleteUserQuery := fmt.Sprintf(`DROP USER IF EXISTS "%s"`, newUserName)
+	err = r.ExecPostgreSQLQuery(ctx, deleteUserQuery, defaultCreds, clusterName, defaultUserSecretNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *PostgreSQLUserReconciler) ExecPostgreSQLQuery(
+	ctx context.Context,
+	query string,
+	defaultCreds *models.Credentials,
+	clusterName string,
+	defaultUserSecretNamespacedName clusterresourcesv1beta1.NamespacedName,
+) error {
 	serviceName := fmt.Sprintf(models.ExposeServiceNameTemplate, clusterName)
-	host := fmt.Sprintf("%s.%s", serviceName, exposeServiceList.Items[0].Namespace)
+	host := fmt.Sprintf("%s.%s", serviceName, defaultUserSecretNamespacedName.Namespace)
 
 	dbURL := fmt.Sprintf("postgres://%s:%s@%s:%d/%s?target_session_attrs=read-write",
-		defaultUsername, defaultPassword, host, models.DefaultPgDbPortValue, models.DefaultPgDbNameValue)
+		defaultCreds.Username, defaultCreds.Password, host, models.DefaultPgDbPortValue, models.DefaultPgDbNameValue)
 
 	conn, err := pgx.Connect(ctx, dbURL)
 	if err != nil {
 		return fmt.Errorf("cannot establish a connection with a PostgreSQL server, err: %w", err)
 	}
 	defer conn.Close(ctx)
 
-	createUserQuery := fmt.Sprintf(`CREATE USER "%s" WITH PASSWORD '%s'`, newUserName, newPassword)
-	_, err = conn.Exec(ctx, createUserQuery)
+	_, err = conn.Exec(ctx, query)
 	if err != nil {
-		return fmt.Errorf("cannot execute creation user query in postgresql, err: %w", err)
+		return fmt.Errorf("cannot execute query in PostgreSQL, err: %w", err)
 	}
 
 	return nil
 }
 
+func (r *PostgreSQLUserReconciler) getDefaultPostgreSQLUserCreds(
+	ctx context.Context,
+	defaultUserSecretNamespacedName clusterresourcesv1beta1.NamespacedName,
+) (*models.Credentials, string, error) {
+	defaultUserSecret := &k8sCore.Secret{}
+
+	namespacedName := types.NamespacedName{
+		Namespace: defaultUserSecretNamespacedName.Namespace,
+		Name:      defaultUserSecretNamespacedName.Name,
+	}
+	err := r.Get(ctx, namespacedName, defaultUserSecret)
+	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			return nil, "", fmt.Errorf("cannot get default PostgreSQL user secret, user reference: %v, err: %w", defaultUserSecretNamespacedName, err)
+		}
+
+		return nil, "", err
+	}
```

Comment:
I think it could be simplified
```suggestion
	if err != nil {
		return nil, "",  fmt.Errorf("cannot get default PostgreSQL user secret, user reference: %v, err: %w", defaultUserSecretNamespacedName, err)
	}
```

Reply:
fixed